### PR TITLE
Hotjar on archive page

### PR DESCRIPTION
**Pull request by ishouty**
* Remove Hotjar from works and images 
* add this only for archive view in Work page
* update HotJar js to typescript 
* update default toggle

### CODE REVIEW

File: common/hooks/useHotjar.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+const useHotjar = (
+  toggles: { [key: string]: boolean },
+  conditionToInsert: boolean
+): void => {
+  const { isHotjarActive } = toggles;
   useEffect(() => {
     isHotjarActive &&
-      (function(h, o, t, j, a, r) {
+      conditionToInsert &&
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (ishouty) [inline common/hooks/useHotjar.ts:11]: had to add this in as there is some types unknown

File: common/hooks/useHotjar.ts [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+      (function(h: any, o, t, j, a, r) {
         h.hj =
           h.hj ||
           function() {
+            // eslint-disable-next-line prefer-rest-params
             (h.hj.q = h.hj.q || []).push(arguments);
```
Comment (ishouty) [inline common/hooks/useHotjar.ts:17]: arguments is not declared which is the default hotjar

File: catalogue/webapp/components/Work/Work.tsx [hunk 3]
```diff
@@ -66,7 +67,8 @@ const Work: FunctionComponent<Props> = ({
   });
 
   const isInArchive = work.parts.length > 0 || work.partOf.length > 0;
-
+  // added condition isInArchive as its a hook as we only want to trigger in that view
+  useHotjar(globalContextData.toggles, isInArchive);
```
Comment (jamesgorrie) [inline catalogue/webapp/components/Work/Work.tsx:71]: By doing it like this - I wonder what the point is of having it as a toggle?

To simplify we could also have the `useHotjar` take no arguments, and just:`
```ts
if (isInArchive) {
  useHotjar()
}
```

Comment (ishouty) [reply catalogue/webapp/components/Work/Work.tsx:71]: got a point there, i would still have to put the condition inside the function as eslint complains about conditions for functional components.

Comment (jamesgorrie) [reply catalogue/webapp/components/Work/Work.tsx:71]: > complains about conditions for functional components.

For function components or for hooks?

Comment (ishouty) [reply catalogue/webapp/components/Work/Work.tsx]: hooks, but we talked about changing solution to useState. nice !

**Review by jamesgorrie — APPROVED** [review]
Nice.

**Pull request merged**